### PR TITLE
drop python3.6 support

### PR DESCRIPTION
**Pull request by samuelcolvin**
## Change Summary

Drop support for python3.6 which has reached EOL.

[This](https://pypistats.org/packages/pydantic) suggests that python 3.6 still makes up around 5% of downloads, but that should  drop quickly now, and you have to break some eggs to make an omlet.

## Related issue number

Don't remember what issue we discussed this on, but we previously agreed to drop python 3.6 suport after v1.9 was released.

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by samuelcolvin** [comment]
@PrettyWood would love your opinion if you have some time.

File: pydantic/typing.py [hunk 1]
```diff
@@ -104,7 +77,8 @@ def get_all_type_hints(obj: Any, globalns: Any = None, localns: Any = None) -> A
 
     def get_origin(t: Type[Any]) -> Optional[Type[Any]]:
         if type(t).__name__ in AnnotatedTypeNames:
-            return cast(Type[Any], Annotated)  # mypy complains about _SpecialForm in py3.6
+            # weirdly this is a runtime requirement, as well as for mypy
```
Comment (PrettyWood) [inline pydantic/typing.py:80]: super weird 🤔

**Review by PrettyWood — APPROVED** [review]
LGTM! Always nice to have negative PRs with some clean up :)

File: pydantic/typing.py (hunk 1 above)
Comment (samuelcolvin) [reply pydantic/typing.py:80]: Ye, things like this scare me a little :fearful:.

**Pull request merged**